packages/teams/src/share-or-unshare-items-to-team.ts

```diff
@@ -0,0 +1,47 @@
+import { ArcGISRequestError, IRequestOptions } from "@esri/arcgis-rest-request";
+import { IItem } from "@esri/arcgis-rest-types";
+import { shareItemToGroups, unshareItemFromGroups } from "@esri/hub-common";
+
+/**
+ * Process an array of items, either sharing them to the group
+ * or unsharing them. Defaults to sharing items. Returns an object comprised of
+ * array of successes, and failures/errors.
+ *
+ * @export
+ * @param {string} groupId | group we are sharing/unsharing to
+ * @param {IItem[]} items | array of items we are sharing/unsharing
+ * @param {IRequestOptions} requestOptions | Auth
+ * @param {("share" | "unshare")} [action="share"] | share or unshare, defaults to share
+ * @return {*}
+ */
+export async function shareOrUnshareItemsToTeam(
+  groupId: string,
+  items: IItem[],
+  requestOptions: IRequestOptions,
+  action: "share" | "unshare" = "share"
+) {
+  // Choose to share or unshare
+  const fnName = action === "share" ? shareItemToGroups : unshareItemFromGroups;
+  const result: {
+    successes: IItem[];
+    failures: IItem[];
+    errors: ArcGISRequestError[];
+  } = {
+    successes: [],
+    failures: [],
+    errors: [],
+  };
+
+  const promises = items.map(async (item) => {
+    try {
+      await fnName(groupId, [item.id], requestOptions);
```

Comment:
This error handling only handles the case where the server responds with a non-200 level response, though the share/unshare calls can technically fail but return a successful 200 response. When this occurs the share response will include a `notSharedWith` Array containing the id of the item that was not shared and the unshare response will include a `notUnsharedFrom` Array containing the id of the item that was not unshared.